### DISCUSSION TITLE
Can I use a different async-tls runtime for rocket sqlx?

### DISCUSSION THREAD

**Discussion by aadityadhruv** [rwf2/Rocket#2390] [Questions]
I am currently trying to connect to a remote MySQL database (hosted on GCP) through rocket. The DB url `Rocket.toml` is in the following format:

```url =  "mysql://user:pass@IP:3306/db_name```

However, I am facing an `InvalidDNSNameError` error while trying to run. Looking into this, it seems its an issue with rustls not supporting direct IPs (Causing the DNS error (https://github.com/launchbadge/sqlx/issues/1095)). Looks like a possible solution is to change the runtime to `native-tls`. Is there a way to do this with rocket's sqlx?

**Comment by aadityadhruv** [comment]
As an update - I got my connection working by changing the db_pools `sqlx` feature from `runtime-tokio-rustls` to `runtime-async-std-native-tls` in the following `db_pools` Cargo.toml. 

EDIT: Please use `runtime-tokio-native-tls` instead. Sergio has explained the reason below.

**Comment by SergioBenitez** [comment]
> As an update - I got my connection working by changing the db_pools `sqlx` feature from `runtime-tokio-rustls` to `runtime-async-std-native-tls` in the following `db_pools` Cargo.toml. 

Note that combining two different async runtimes (`tokio` and `async-std`) is not recommended and can lead to silent issues. Use the appropriate `tokio` version instead: `runtime-tokio-native-tls`.

**Reply by aadityadhruv** [reply]
Yup, you did mention this yesterday on the Matrix group. I forgot to update this comment with that. Thanks!
